src/main.ts

```diff
@@ -72,7 +74,12 @@ async function get_rustup(toolchain: string): Promise<string> {
 
 async function run() {
     const opts = args.toolchain_args();
-    const rustup = await get_rustup(opts.name);
+    const profile = opts.profile ? opts.profile : 'default';
+    const rustup = await get_rustup(opts.name, profile);
+
+    if (opts.profile) {
+        await exec.exec(rustup, ['set', 'profile', opts.profile]);
```

Comment:
It is necessary to run self-update before setting the profile so that the older rustup does not emit an error (discussed in https://github.com/actions-rs/toolchain/pull/15#issuecomment-542547552):

```suggestion
        await exec.exec(rustup, ['self', 'update']);
        await exec.exec(rustup, ['set', 'profile', opts.profile]);
```

If we want to limit the self-updating before 1.20.0, additional version check (including semver parsing) is required at here.